File: src/main/kotlin/com/geonyeong/insurance/entity/GuideEntity.java
package com.geonyeong.insurance.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.util.List;

@Entity
@Table(name="guide")
@Getter
@Setter
public class GuideEntity {
    @Id
    private @Column(name = "no") Long no;

    private @Column(name = "title") String title;

    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "kind_company_id")
    @JsonBackReference
    private KindCompanyEntity kindCompanyEntity;

}

File: src/main/kotlin/com/geonyeong/insurance/entity/GuideEntity.kt
package com.geonyeong.insurance.entity

import com.fasterxml.jackson.annotation.JsonBackReference
import javax.persistence.*

@Entity
@Table(name = "guide")
data class GuideEntity(
        @Id
        @Column(name = "no")
        var no: Long?,
        @Column(name = "title")
        var title:String?,
        @ManyToOne(cascade = arrayOf(CascadeType.ALL))
        @JoinColumn(name = "kind_company_id")
        @JsonBackReference
        var kindCompanyEntity: KindCompanyEntity?
){
        constructor():this(0,"",null)
}